standards/core/rln-contract.md

```diff
@@ -0,0 +1,199 @@
+
+The user can extend the membership both during and after its grace period.
+By not extending the membership before the grace period expires, the user assumes the risk of the membership being taken over at any moment.
+
+A user can hold multiple memberships.
+Memberships are not transferable.
+There is no limitation on the number of membership per user (apart from the global limits, see parameter table).
+
+## Contract governance and mutability
+
+The contract governance is as follows:
+
+1. the first deployment of the contract allows the _Owner_ to modify certain parameters (TBD) under certain constraints (TBD);
+2. at some point, the _Owner_ SHOULD renounce their privileges, and the contract MUST become immutable;
+3. further upgrades, if necessary, can be done by deploying a new contract and migrating the membership set.
+
+## Parameters
+
+We make the following design choices:
+- there are `3` membership tiers: low-, mid-, and high-tier;
+- prices are quoted in `USD`;
+- payment is accepted in `DAI`;
+- the price for a given tier is linearly proportional to its rate limit.
+
+For the initial deployment, we suggest the following values.
+
+| Parameter                                                 | Symbol     | Value   | Units                                                                                | Modifiable? | Comment                        |
+| --------------------------------------------------------- | ---------- | ------- | ------------------------------------------------------------------------------------ | ----------- | ------------------------------ |
+| Epoch length                                              | `epoch`    | `10`    | minutes                                                                              | Yes         |                                |
+| Maximum total rate limit of concurrent active memberships | `R_{max}`  | TBD     | messages per `epoch`                                                                 | Yes         | Can be replaced with `N_{max}` |
+| Maximum number of concurrent active memberships           | `N_{max}`  | `10000` | units                                                                                | Yes         | Can be replaced with `R_{max}` |
+| Rate limit for low-tier                                   | `R_{low}`  | `20`    | messages per `epoch`                                                                 | TBD         |                                |
+| Rate limit for mid-tier                                   | `R_{mid}`  | `200`   | messages per `epoch`                                                                 | TBD         |                                |
+| Rate limit for high-tier                                  | `R_{high}` | `600`   | messages per `epoch`                                                                 | TBD         |                                |
```

Comment:
What do you mean by progressive? I see two ways to interpret this:
1. any continuous function that calculates membership price depending on a user-defined rate limit;
2. same as above _and_ the function not simply multiplication by a constant (i.e., bulk discounts for memberships with higher limits, etc).

In other words, would a linear function satisfy your definition of progressive?

Reply:
> What do you mean by progressive?

Yes, I mean giving the user the option to set up an arbitrary limit. It could be linear and top-bounded. The more the user pays the higher rate limit is allowed.